framework/include/mesh/MooseMesh.h

```diff
@@ -1309,6 +1309,12 @@ class MooseMesh : public MooseObject, public Restartable, public PerfGraphInterf
    */
   MooseAppCoordTransform & coordTransform();
 
+  /**
+   * @return whether coordinate transformation exists by having rotation parameters and
+   * up_direction parameter for the opposite of the gravity vector.
+   */
+  bool hasCoordTransform() const;
```

Comment:
a scaling is a coordinate transformation as well.
Please follow the current design
```
  /**
   * Returns true if the app has scaling and/or rotation transformation
   */
  bool hasScalingOrRotationTransformation() const;
```
with for example: hasNonScalingTransformation()

and place this declaration next to it, in MooseAppCoordTransform